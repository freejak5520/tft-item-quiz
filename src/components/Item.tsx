import Image from "next/image";

type Props = {
  item?: Item;
  onClick?: React.MouseEventHandler<HTMLDivElement>;
  active?: boolean;
  size?: number;
};

const Item = ({ item, onClick, active = false, size = 75 }: Props) => {
  return (
    <div
      className={`relative flex flex-col items-center overflow-hidden rounded border-2 border-gray-600 p-0 shadow ${
        onClick && "cursor-pointer"
      }`}
      onClick={onClick}
    >
      {active && (
        <div className="absolute left-0 top-0 h-full w-full border-2 border-green-700 bg-black bg-opacity-50"></div>
      )}
      {item ? (
        <Image
          src={`/img/items/${item.image}`}
          alt={item.name}
          width={size}
          height={size}
        />
      ) : (
        <Image
          src={"/img/items/none.png"}
          width={size}
          height={size}
          alt="none"
        />
      )}
      {/* <div className="text-xs">{item.name}</div> */}
    </div>
  );
};

export default Item;
