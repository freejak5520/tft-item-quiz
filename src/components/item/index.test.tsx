import { render } from "@testing-library/react";
import Item from ".";

describe("Item", () => {
  it("renders the item", () => {
    const { container } = render(<Item />);

    expect(container).toMatchSnapshot();
  });
});
