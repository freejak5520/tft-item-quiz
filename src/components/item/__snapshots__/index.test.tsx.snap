// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders the item 1`] = `
<div>
  <div
    class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow undefined"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="75"
      loading="lazy"
      src="/_next/image?url=%2Fimg%2Fitems%2Fgray.webp&w=256&q=75"
      srcset="/_next/image?url=%2Fimg%2Fitems%2Fgray.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2Fgray.webp&w=256&q=75 2x"
      style="color: transparent;"
      width="75"
    />
  </div>
</div>
`;
